src/ucp/rndv/rndv_ats.c

```diff
@@ -13,45 +13,39 @@
 
 static void ucp_proto_rndv_ats_probe(const ucp_proto_init_params_t *init_params)
 {
-    ucp_proto_caps_t caps, input_caps;
+    ucp_proto_common_init_params_t params;
     ucp_proto_rndv_ack_priv_t priv;
-    ucp_proto_perf_range_t *range0;
+    ucp_proto_perf_t *perf;
     ucs_status_t status;
 
-    if (ucp_proto_rndv_init_params_is_ppln_frag(init_params)) {
-        return;
-    }
-
-    input_caps.min_length = 0;
-    input_caps.num_ranges = 1;
-    range0                = &input_caps.ranges[0];
-    range0->node          = NULL;
-    ucp_proto_perf_set(range0->perf, UCS_LINEAR_FUNC_ZERO);
+    params              = ucp_proto_common_init_params(init_params);
+    params.cfg_priority = 80;
 
     /* This protocols supports either a regular rendezvous receive but without
      * data, or a rendezvous receive which should ignore the data.
      * In either case, we just need to send an ATS.
      */
     if (ucp_proto_init_check_op(init_params, UCS_BIT(UCP_OP_ID_RNDV_RECV))) {
-        range0->max_length = 0;
+        params.max_length = 0;
     } else if (ucp_proto_init_check_op(init_params,
                                        UCS_BIT(UCP_OP_ID_RNDV_RECV_DROP))) {
-        range0->max_length = SIZE_MAX;
+        params.max_length = SIZE_MAX;
     } else {
         return;
     }
 
-    status = ucp_proto_rndv_ack_init(init_params, UCP_PROTO_RNDV_ATS_NAME,
-                                     &input_caps, UCS_LINEAR_FUNC_ZERO, &priv,
-                                     &caps);
-    ucp_proto_select_caps_cleanup(&input_caps);
-
-    if (status != UCS_OK) {
+    status = ucp_proto_rndv_ack_init(&params, UCP_PROTO_RNDV_ATS_NAME,
+                                     UCS_LINEAR_FUNC_ZERO, &perf, &priv);
+    if ((status != UCS_OK) || (priv.lane == UCP_NULL_LANE)) {
```

Comment:
if priv.lane == UCP_NULL_LANE, shall we release perf? or assert it is NULL?

Reply:
The only case when status is OK and priv.lane is NULL is case when PPLN is detected. I changed this condition to `perf == NULL`.